projects/step-core/src/lib/modules/bookmarks/injectables/bookmark.service.ts

```diff
@@ -0,0 +1,27 @@
+import { inject, Injectable } from '@angular/core';
+import { BehaviorSubject, map } from 'rxjs';
+import { AugmentedBookmarksService } from '../../../client/augmented/services/augmented-bookmarks.service';
+import { switchMap } from 'rxjs/operators';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class BookmarkService {
+  private _bookmarksApi = inject(AugmentedBookmarksService);
+
+  public refreshBookmarks$ = new BehaviorSubject<unknown>(undefined);
```

Comment:
1. fields without modifier are public by default
2. Defining field like this, wont prevent someone to do the following:
```
private _bookmarksService = inject(BookmarkService);
...
this._bookmarksService.refreshBookmarsk$ = undefined;
```
It will broke the service. `readonly` modifier makes more sense in this case
3. But subject existence looks like internal service logic, that can be encapsulated. It can be achieved by introducing a separate method, that will invoke the trigger:
```
private refreshBookmarks$ = new BehaviorSubject<unknow>(undefined);

readonly bookmars$ = this.refreshBookmarks$.pipe(
 ....
);

refreshBookmarks(): void {
    this.refreshBookmarks$.next(undefined);
}
```

Reply:
Subject changed to private and refreshBookmarks method added